course6/toyapp-adam/util/keyboard.cc

```diff
@@ -0,0 +1,68 @@
+#include "keyboard.h"
+
+#include <cassert>
+
+namespace Util {
+
+void Keyboard::Press(KeyCode keyCode)
+{
+	assert(keyCode >= 0);
+	assert(keyCode < NUM_KEYCODES);
+
+	mKeys[keyCode].up = 0;
+	mKeys[keyCode].down = 1;
+	mKeys[keyCode].pressed = 1;
+}
+
+void Keyboard::Release(KeyCode keyCode)
+{
+	assert(keyCode >= 0);
+	assert(keyCode < NUM_KEYCODES);
+
+	mKeys[keyCode].up = 1;
+	mKeys[keyCode].down = 0;
+	mKeys[keyCode].released = 1;
+}
+
+void Keyboard::ClearChanges()
+{
+	for(KeyCode i = 0; i < NUM_KEYCODES; ++i)
```

Comment:
Use range-based for

Reply:
Neat, I did not know it works for plain arrays, thanks!